web/packages/teleport/src/Users/UserAddEdit/TraitsEditor.tsx

```diff
@@ -0,0 +1,271 @@
+                      label: r,
+                    }))}
+                    label="Value"
+                    isMulti
+                    isSearchable
+                    isClearable={false}
+                    value={traitValues}
+                    rule={requiredField('Trait value cannot be empty')}
+                    onChange={e => {
+                      handleInputChange({
+                        option: e,
+                        labelField: 'traitValues',
+                        index: index,
+                      });
+                    }}
+                    formatCreateLabel={(i: string) =>
+                      'Trait value: ' + `"${i}"`
+                    }
+                    isDisabled={attempt.status === 'processing'}
+                  />
+                </Box>
+                <ButtonIcon
+                  ml={1}
+                  mt={7}
+                  size={1}
+                  title="Remove Trait"
+                  aria-label="Remove Trait"
+                  onClick={() => removeTraitPair(index)}
+                  css={`
+                    &:disabled {
+                      opacity: 0.65;
+                      pointer-events: none;
+                    }
+                  `}
+                  disabled={attempt.status === 'processing'}
+                >
+                  <Trash size="medium" />
+                </ButtonIcon>
+              </Flex>
+            </Box>
+          );
+        })}
+      </Box>
+
+      <Box mt={5}>
+        <ButtonBorder
+          onClick={addNewTraitPair}
+          label={addLabelText}
+          css={`
+            padding-left: 12px;
+            &:disabled {
+              .icon-add {
+                opacity: 0.35;
+              }
+              pointer-events: none;
+            }
+          `}
+          disabled={attempt.status === 'processing'}
+        >
+          <Add
+            className="icon-add"
+            size={12}
+            css={`
+              margin-right: 3px;
+            `}
+          />
+          {addLabelText}
+        </ButtonBorder>
+      </Box>
+    </Box>
+  );
+}
+
+type InputOption = {
+  labelField: 'traitKey' | 'traitValues';
+  option: Option | Option[];
+  index: number;
+};
+
+const requireNoDuplicateTraits =
+  (configuredTraits: TraitsOption[]) => (enteredTrait: Option) => () => {
+    let k = configuredTraits.map(trait => trait.traitKey.value.toLowerCase());
+    let occurance = 0;
+    k.forEach(key => {
+      if (key === enteredTrait.value.toLowerCase()) {
+        occurance++;
+      }
+    });
+    if (occurance > 1) {
+      return { valid: false, message: 'Trait key should be unique for a user' };
+    }
+    return { valid: true };
+  };
+
+export function traitsToTraitsOption(allTraits: AllUserTraits): TraitsOption[] {
+  let newTrait = [];
+  for (let trait in allTraits) {
+    if (!allTraits[trait][0]) {
+      continue;
+    }
+    if (allTraits[trait].length > 0) {
```

Comment:
> also is it valid to add a trait with a empty name?

It shouldn't be a valid case but I know the `tctl` does not check this.

The `if (!allTraits[trait][0]) {` block exists because when user is created using tctl, some trait gets default `null` value and some, such as `host_user_gid` gets `""`. In the latter case, the value is essentially not a user configured value so I think it we don't need show them in traits editor as a "pre-configured" traits. 

<img width="718" alt="Screenshot 2024-06-13 at 12 25 02 PM" src="https://github.com/gravitational/teleport/assets/10177602/398d3590-8b49-4bcc-beac-978e618e93a1">


Pinging @r0mant if there is any reason to allow traits with empty value.

Reply:
Can't think of any.